keras_hub/src/layers/modeling/transformer_encoder.py

```diff
@@ -207,22 +207,13 @@ def call(
         if self.normalize_first:
             x = self._self_attention_layer_norm(x)
 
-        if return_attention_scores:
-            x, attention_scores = self._self_attention_layer(
-                query=x,
-                value=x,
-                attention_mask=self_attention_mask,
-                return_attention_scores=return_attention_scores,
-                training=training,
-            )
-            return x, attention_scores
-        else:
-            x = self._self_attention_layer(
-                query=x,
-                value=x,
-                attention_mask=self_attention_mask,
-                training=training,
-            )
+        x, attention_scores = self._self_attention_layer(
+            query=x,
+            value=x,
+            attention_mask=self_attention_mask,
+            return_attention_scores=True,
```

Comment:
I was looking to see how `return_attention_scores` is used in MHA and looking at [this](https://github.com/keras-team/keras/blob/8409e18dd73dcffaa4c5772326ce33866d8c4df5/keras/src/layers/attention/multi_head_attention.py#L433), I was thinking if setting `return_attention_scores=True` here could cause any problems. If we set it `True` here, the user doesn't have a way to disable it if they want to use flash attention, right?

Reply:
Ah, good spot. I'll keep the if...else, then

Reply:
I think we can still get rid of the `if-else` block by directly passing the flag to MHA (setting it like this `return_attention_scores=return_attention_scores`). We don't have to pass the flag only when it's `True`. wdyt?

Reply:
You're right! Let's keep it the way it is then!